geode-core/src/main/java/org/apache/geode/internal/cache/control/HeapMemoryMonitor.java

```diff
@@ -144,82 +84,34 @@
   private final ResourceAdvisor resourceAdvisor;
   private final InternalCache cache;
   private final ResourceManagerStats stats;
-  private final TenuredHeapConsumptionMonitor tenuredHeapConsumptionMonitor;
 
-  @MutableForTesting
-  private static boolean testDisableMemoryUpdates = false;
   @MutableForTesting
   private static long testBytesUsedForThresholdSet = -1;
 
-  /**
-   * Determines if the name of the memory pool MXBean provided matches a list of known tenured pool
-   * names.
-   *
-   * Package private for testing.
-   * checkTenuredHeapConsumption
-   *
-   * @param memoryPoolMXBean The memory pool MXBean to check.
-   * @return True if the pool name matches a known tenured pool name, false otherwise.
-   */
-  static boolean isTenured(MemoryPoolMXBean memoryPoolMXBean) {
-    if (memoryPoolMXBean.getType() != MemoryType.HEAP) {
-      return false;
-    }
-
-    String name = memoryPoolMXBean.getName();
-
-    return name.equals("CMS Old Gen") // Sun Concurrent Mark Sweep GC
-        || name.equals("PS Old Gen") // Sun Parallel GC
-        || name.equals("G1 Old Gen") // Sun G1 GC
-        || name.equals("Old Space") // BEA JRockit 1.5, 1.6 GC
-        || name.equals("Tenured Gen") // Hitachi 1.5 GC
-        || name.equals("Java heap") // IBM 1.5, 1.6 GC
-        || name.equals("GenPauseless Old Gen") // azul C4/GPGC collector
-
-        // Allow an unknown pool name to monitor
-        || (HEAP_POOL != null && name.equals(HEAP_POOL));
-  }
+  @Immutable
+  private static final ServiceLoader<HeapUsageProvider> heapUsageMonitorLoader =
+      ServiceLoader.load(HeapUsageProvider.class);
 
   HeapMemoryMonitor(final InternalResourceManager resourceManager, final InternalCache cache,
-      final ResourceManagerStats stats,
-      TenuredHeapConsumptionMonitor tenuredHeapConsumptionMonitor) {
+      final ResourceManagerStats stats) {
     this.resourceManager = resourceManager;
     resourceAdvisor = (ResourceAdvisor) cache.getDistributionAdvisor();
     this.cache = cache;
     this.stats = stats;
-    this.tenuredHeapConsumptionMonitor = tenuredHeapConsumptionMonitor;
+    heapUsageMonitor = findHeapUsageMonitor();
+    thresholds = new MemoryThresholds(heapUsageMonitor.getMaxMemory());
+    mostRecentEvent = new MemoryEvent(ResourceType.HEAP_MEMORY,
+        MemoryState.DISABLED, MemoryState.DISABLED, null, 0L, true, thresholds);
   }
 
-  /**
-   * Returns the tenured pool MXBean or throws an IllegaleStateException if one couldn't be found.
-   */
-  public static MemoryPoolMXBean getTenuredMemoryPoolMXBean() {
-    if (tenuredMemoryPoolMXBean != null) {
-      return tenuredMemoryPoolMXBean;
+  private static HeapUsageProvider findHeapUsageMonitor() {
```

Comment:
I don't really like "get" either since to me that makes it seem like it is returning a value that already exists. How about "createHeapUsageProvider"?

Reply:
Maybe we take a leaf out of the ServiceLoaders and call it "load"...